src/main/java/com/regnosys/testing/schemeimport/SchemeImporterTestHelper.java

```diff
@@ -0,0 +1,155 @@
+import com.regnosys.rosetta.common.util.ClassPathUtils;
+import com.regnosys.rosetta.common.util.CollectionUtils;
+import com.regnosys.rosetta.common.util.UrlUtils;
+import com.regnosys.rosetta.rosetta.RosettaEnumValue;
+import com.regnosys.rosetta.rosetta.RosettaEnumeration;
+import com.regnosys.rosetta.rosetta.RosettaModel;
+import com.regnosys.rosetta.transgest.ModelLoader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+import static org.junit.Assert.assertTrue;
+
+
+public class SchemeImporterTestHelper {
+
+    public enum EnumComparison {
+        /**
+         * Model and coding scheme enum values exactly match
+         */
+        ExactMatch,
+        /**
+         * Model contains all coding scheme enum values, however additional values are allowed
+         */
+        AdditiveMatch
+    }
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchemeImporterTestHelper.class);
+    @Inject
+    private SchemeImporter schemeImporter;
+    @Inject
+    private ModelLoader modelLoader;
+    private static final Comparator<RosettaEnumValue> enumValueComparator = Comparator.comparing(RosettaEnumValue::getName)
+            .thenComparing(RosettaEnumValue::getDefinition);
+
+    public void checkEnumsAreValid(String rosettaPathRoot, String body, String codingScheme, SchemeEnumReader schemeEnumReader, boolean writeTestOutput, EnumComparison enumComparison) throws IOException {
+        URL[] rosettaPaths = getRosettaPaths(rosettaPathRoot);
+
+        List<RosettaModel> models = modelLoader.loadRosettaModels(rosettaPaths);
+        List<RosettaEnumeration> rosettaEnumsFromModel = schemeImporter.getRosettaEnumsFromModel(models, body, codingScheme);
+
+        if(writeTestOutput) {
+            persistEnumValues(rosettaEnumsFromModel, schemeEnumReader, enumComparison);
+        }
+
+        validateEnumValues(rosettaEnumsFromModel, schemeEnumReader, enumComparison);
+
+
+
+        //persist enum values
+        //if it is exact match
+            //Blat the stuff for Exact match and replace from schema enums
+        //if it is additive match
+            //if coding scheme has extra values, add them alone
+        //Build a list of rosetta enum values combinig but removing duplicates
+
+        //loop through coding schme and remove it from model based on the name - there will be a pure list and then we can add the extra ones from coding schmes at the bottom
+    }
+
+    private void persistEnumValues(List<RosettaEnumeration> rosettaEnumsFromModel, SchemeEnumReader schemeEnumReader, EnumComparison enumComparison) throws IOException {
+        Map<String, String> generatedFromScheme = null;
+        if(enumComparison.equals(EnumComparison.ExactMatch)){
```

Comment:
Enums should be compared with an `==`, e.g. `enumComparison == EnumComparison.ExactMatch`

Reply:
But actually this would be cleaner to do with a switch statement:

```
switch (enumComparison) {
    case ExactMatch:
        // TODO add your code
    case AdditiveMatch:
        // TODO throw unsupported exception?
    default:
        throw new IllegalArgumentException("Unknown enum value " + enumComparison);
}
```